### PR TITLE
Fix CI error for service session affinity

### LINKED ISSUE #53741: [e2e failure] "should function for client IP based session affinity" constantly fails
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/priority failing-test
/sig network

**What happened**:
From https://k8s-testgrid.appspot.com/google-gce#gci-gce-slow&sort-by-failures=&width=5,
`Services should function for client IP based session affinity: udp/http` seems to be broken right after they were added by https://github.com/kubernetes/kubernetes/pull/50447.

Example error message:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:225
Oct 11 16:39:42.406: Expect endpoints: map[], got: map[netserver-1:{}]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:222
```
and
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:249
Oct 11 16:37:49.106: test session affinity, cost time: 1m31.742990485s
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:248
```

**Anything else we need to know?**:
@m1093782566 Is it as expected?

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Fix CI error for service session affinity -- debug

**Which issue this PR fixes**: 

fixes #53741

**Special notes for your reviewer**:

I remove the [slow] tag so that these test cases can be run in PR request. We may need to add back the [slow] tag when this PR is ready to get in.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by MrHohn** [comment]
Will soon take a look.
/assign
/unassign @danwinship @dcbw

**Comment by m1093782566** [comment]
ping @MrHohn ?

File: test/e2e/framework/networking_utils.go [hunk 1]
```diff
@@ -255,10 +256,15 @@ func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containe
...
+			// Return immediately when we successfully fetch endpoints
+			if len(eps) > 0 {
+				return eps, nil
+			}
+			// TODO: get rid of this delay #36281
+			time.Sleep(hitEndpointRetryDelay)
 		}
 	}
-	return nil, fmt.Errorf("Failed to get endpoints:\nTries %d\nCommand %v\n", minTries, cmd)
+	return nil, fmt.Errorf("error getting endpoints:\nTries %d\nCommand %v\n", minTries, cmd)
```
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: Seems we are not really using `minTries` in this fuction? Could we remove `minTries` and rename `maxTries` to `tries`?

File: test/e2e/framework/networking_utils.go [hunk 2]
```diff
@@ -255,10 +256,15 @@ func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containe
 					eps.Insert(trimmed)
 				}
 			}
-			return eps, nil
+			// Return immediately when we successfully fetch endpoints
+			if len(eps) > 0 {
```
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: Could be great if we document what exactly `GetEndpointsFromContainer()`  is for. This use case seem to be tricky.

**Review by MrHohn — APPROVED** [review]
Thanks for the fix, overall LGTM.

File: test/e2e/framework/networking_utils.go (hunk 2 above)
Comment (m1093782566) [reply test/e2e/framework/networking_utils.go]: It's fixed now. PTAL.

File: test/e2e/framework/networking_utils.go [hunk 3]
```diff
@@ -220,11 +220,16 @@ func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, tar
...
-	return config.GetEndpointsFromContainer(protocol, config.TestContainerPod.Status.PodIP, targetIP, TestContainerHttpPort, targetPort, maxTries, minTries)
+func (config *NetworkingTestConfig) GetEndpointsFromTestContainer(protocol, targetIP string, targetPort, tries int) (sets.String, error) {
+	return config.GetEndpointsFromContainer(protocol, config.TestContainerPod.Status.PodIP, targetIP, TestContainerHttpPort, targetPort, tries)
 }
 
-func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containerIP, targetIP string, containerHttpPort, targetPort, maxTries, minTries int) (sets.String, error) {
+// GetEndpointsFromContainer executes a curl via kubectl exec in a test container,
+// which might then translate to a tcp or udp request based on the protocol argument
+// in the url.
+// - maxTries is the maximum number of curl attempts. If this many attempts pass and
```
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: Actually, `maxTries` also makes sense, do you mind either changing the `tries` in funciton back to `maxTries` or change this comment to `tries`?

**Review by MrHohn — APPROVED** [review]
Thanks, LGTM, could you squash commits?

File: test/e2e/framework/networking_utils.go (hunk 3 above)
Comment (m1093782566) [reply test/e2e/framework/networking_utils.go]: I changed the comment to `tries`. PTAL. Thanks!

**Comment by MrHohn** [comment]
Thanks for the fix!

/lgtm

**Comment by MrHohn** [comment]
/priority failing-test 
/assign @bowei 
For approval.

**Comment by MrHohn** [comment]
/unassign @bowei 

Actually, figured @thockin might be the right person to approve this :)

/assign @thockin 
for approval.

**Comment by m1093782566** [comment]
ping @thockin ~

**Comment by m1093782566** [comment]
Thanks!

**Pull request merged**